<launch>
    <!-- IMU relay from lingao_msgs/Imu to sensor_msgs/Imu -->
    <node pkg="imu_calib" type="apply_calib" name="apply_calib" output="screen" respawn="false">
        <param name="calib_file" value="$(find lingao_bringup)/param/imu/imu_calib.yaml" />
        <param name="calibrate_gyros" value="true" />
        <param name="gyro_calib_samples" value="300" />
    </node>

    <!-- Filter and fuse raw imu data -->
    <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_madgwick" output="screen" respawn="false" >
        <param name="fixed_frame" value="base_link" />
        <param name="use_mag" value="false" />
        <param name="gain" value="0.05" />
        <param name="publish_tf" value="false" />
        <param name="use_magnetic_field_msg" value="false" /> 
        <param name="world_frame" value="enu" />
        <param name="orientation_stddev" value="0.1" />
        <param name="publish_debug_topics" value="false" />

        <remap from="imu/data_raw" to="/imu/data_raw_calib" />
    </node>
        
    <!-- Publish static transform from base_footprint to imu_link -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_footprint_to_imu_link" args="0 0 0.05 0 0 0  /base_link /imu_link"/> -->
</launch>
